spyder/widgets/variableexplorer/tests/test_dataframeeditor.py

```diff
@@ -263,5 +265,72 @@ def test_sort_dataframe_with_category_dtypes(qtbot):  # cf. issue 5361
     assert data(dfm, 1, 1) == 'category'
 
 
+def test_dataframemodel_set_data_overflow(monkeypatch):
+    """Unit test #6114: entry of an overflow int caught and handled properly"""
+    MockQMessageBox = Mock()
+    attr_to_patch = ('spyder.widgets.variableexplorer' +
+                     '.dataframeeditor.QMessageBox')
+    monkeypatch.setattr(attr_to_patch, MockQMessageBox)
+
+    if system() == 'Linux':
+        int32_bit_exponent = 66
+    else:
+        int32_bit_exponent = 34
+    test_parameters = [(1, numpy.int32, int32_bit_exponent),
+                       (2, numpy.int64, 66)]
+    for idx, int_type, bit_exponent in test_parameters:
+        test_df = DataFrame(numpy.arange(7, 11), dtype=int_type)
+        model = DataFrameModel(test_df.copy())
+        index = model.createIndex(2, 1)
+        assert not model.setData(index, str(int(2 ** bit_exponent)))
+        MockQMessageBox.critical.assert_called_with(ANY, "Error", ANY)
+        assert MockQMessageBox.critical.call_count == idx
+        assert numpy.sum(test_df[0].as_matrix() ==
+                         model.df.as_matrix()) == len(test_df)
+
+
+@flaky(max_runs=3)
+@pytest.mark.skipif(os.environ.get('CI', None) is not None and
+                    system() == "Linux",
```

Comment:
Please use `sys.platform.startswith('linux')` here

Reply:
Sure, that's what I used on the other one previously, but couldn't remember the exact syntax. Just curious, what's the advantage over platform.system()?

Reply:
It's just our custom throughout our entire codebase.

Reply:
Makes sense, thanks.